doc/iam.md

```diff
@@ -0,0 +1,6 @@
+# Ignition and IAM roles
```

Comment:
Any preference on where it should go? `doc/examples.md` doesn't fit because this is an experimental feature, and it doesn't seem to match the style/amount of info on other fields in `doc/configuration-v2_1-experimental.md`.